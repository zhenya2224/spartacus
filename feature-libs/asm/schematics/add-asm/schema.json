{
  "$schema": "http://json-schema.org/schema",
  "$id": "AsmSchematics",
  "title": "ASM Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the asm features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["ASM"]
    }
  },
  "required": []
}
