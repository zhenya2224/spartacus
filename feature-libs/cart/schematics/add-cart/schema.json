{
  "$schema": "http://json-schema.org/schema",
  "$id": "CartSchematics",
  "title": "Cart Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Saved cart feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Saved-Cart", "Quick-Order", "Import-Export"],
        "type": "string"
      },
      "default": ["Saved-Cart", "Quick-Order", "Import-Export"],
      "x-prompt": {
        "message": "Which Cart features would you like to set up?",
        "type": "list",
        "items": [
          {
            "value": "Saved-Cart",
            "label": "Saved Cart"
          },
          {
            "value": "Quick-Order",
            "label": "Quick Order"
          },
          {
            "value": "Import-Export",
            "label": "Import/Export"
          }
        ]
      }
    }
  },
  "required": []
}
