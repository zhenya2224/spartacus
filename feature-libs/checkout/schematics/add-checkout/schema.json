{
  "$schema": "http://json-schema.org/schema",
  "$id": "CheckoutSchematics",
  "title": "Checkout Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Checkout feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Checkout"]
    }
  },
  "required": []
}
