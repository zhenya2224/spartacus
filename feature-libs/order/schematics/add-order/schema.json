{
  "$schema": "http://json-schema.org/schema",
  "$id": "OrderSchematics",
  "title": "Order Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Order feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Order"]
    }
  },
  "required": []
}
