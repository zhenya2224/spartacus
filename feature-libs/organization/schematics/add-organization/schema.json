{
  "$schema": "http://json-schema.org/schema",
  "$id": "OrganizationSchematics",
  "title": "Organization Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the organization features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Administration", "Order-Approval"],
        "type": "string"
      },
      "default": ["Administration", "Order-Approval"],
      "x-prompt": {
        "message": "Which Organization features would you like to set up?",
        "type": "list",
        "items": [
          {
            "value": "Administration",
            "label": "Administration (b2b feature)"
          },
          { "value": "Order-Approval", "label": "Order Approval (b2b feature)" }
        ]
      }
    }
  },
  "required": []
}
