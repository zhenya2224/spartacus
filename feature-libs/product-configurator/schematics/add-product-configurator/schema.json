{
  "$schema": "http://json-schema.org/schema",
  "$id": "ProductConfiguratorSchematics",
  "title": "Product Configurator Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the product configurator features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": [
          "VC-Configurator",
          "CPQ-Configurator",
          "Textfield-Configurator"
        ],
        "type": "string"
      },
      "x-prompt": {
        "message": "Which product configurator features would you like to set up?",
        "type": "list",
        "items": [
          {
            "value": "VC-Configurator",
            "label": "Variant Configurator"
          },
          {
            "value": "CPQ-Configurator",
            "label": "CPQ Configurator (b2b feature)"
          },
          {
            "value": "Textfield-Configurator",
            "label": "Textfield Configurator"
          }
        ]
      }
    }
  },
  "required": []
}
