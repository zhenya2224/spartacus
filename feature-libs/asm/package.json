{
  "name": "@spartacus/asm",
  "version": "4.1.0-next.0",
  "description": "ASM feature library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "asm"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/asm",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^12.0.5",
    "@angular/common": "^12.0.5",
    "@angular/core": "^12.0.5",
    "@angular/forms": "^12.0.5",
    "@ngrx/effects": "^12.1.0",
    "@ngrx/store": "^12.1.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
