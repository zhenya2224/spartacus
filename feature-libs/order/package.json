{
  "name": "@spartacus/order",
  "version": "4.2.0",
  "description": "Order feature library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "order"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/order",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^12.0.5",
    "@angular/common": "^12.0.5",
    "@angular/core": "^12.0.5",
    "@angular/forms": "^12.0.5",
    "@angular/router": "^12.0.5",
    "@ng-bootstrap/ng-bootstrap": "^10.0.0",
    "@ng-select/ng-select": "^7.0.1",
    "@ngrx/effects": "^12.1.0",
    "@ngrx/store": "^12.1.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "@spartacus/styles": "4.1.0-next.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
