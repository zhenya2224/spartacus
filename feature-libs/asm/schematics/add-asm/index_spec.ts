/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  CLI_ASM_FEATURE,
  LibraryOptions as SpartacusAsmOptions,
  SpartacusOptions,
  SPARTACUS_SCHEMATICS,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');
const featureModulePath =
  'src/app/spartacus/features/asm/asm-feature.module.ts';

describe('Spartacus Asm schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: SpartacusAsmOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const asmFeatureOptions: SpartacusAsmOptions = {
    ...libraryNoFeaturesOptions,
    features: [CLI_ASM_FEATURE],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      '../../projects/schematics/src/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        SPARTACUS_SCHEMATICS,
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', libraryNoFeaturesOptions, appTree)
        .toPromise();
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(featureModulePath)).toBeFalsy();
    });

    it('should install necessary Spartacus libraries', () => {
      const packageJson = JSON.parse(appTree.readContent('package.json'));
      let dependencies: Record<string, string> = {};
      dependencies = { ...packageJson.dependencies };
      dependencies = { ...dependencies, ...packageJson.devDependencies };

      for (const toAdd in peerDependencies) {
        // skip the SPARTACUS_SCHEMATICS, as those are added only when running by the Angular CLI, and not in the testing environment
        if (
          !peerDependencies.hasOwnProperty(toAdd) ||
          toAdd === SPARTACUS_SCHEMATICS
        ) {
          continue;
        }
        // TODO: after 4.0: use this test, as we'll have synced versions between lib's and root package.json
        // const expectedVersion = (peerDependencies as Record<
        //   string,
        //   string
        // >)[toAdd];
        const expectedDependency = dependencies[toAdd];
        expect(expectedDependency).toBeTruthy();
        // expect(expectedDependency).toEqual(expectedVersion);
      }
    });
  });

  describe('Asm feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', asmFeatureOptions, appTree)
          .toPromise();
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            {
              ...asmFeatureOptions,
              lazy: false,
            },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });
});
