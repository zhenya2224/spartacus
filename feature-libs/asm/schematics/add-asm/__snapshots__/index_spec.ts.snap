// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Asm schematics: ng-add Asm feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { AsmModule } from \\"@spartacus/asm\\";
import { asmTranslationChunksConfig, asmTranslations } from \\"@spartacus/asm/assets\\";
import { AsmRootModule } from \\"@spartacus/asm/root\\";
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";

@NgModule({
  declarations: [],
  imports: [
    AsmRootModule,
    AsmModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: asmTranslations,
      chunks: asmTranslationChunksConfig,
    },
  })]
})
export class AsmFeatureModule { }
"
`;

exports[`Spartacus Asm schematics: ng-add Asm feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { asmTranslationChunksConfig, asmTranslations } from \\"@spartacus/asm/assets\\";
import { AsmRootModule, ASM_FEATURE } from \\"@spartacus/asm/root\\";
import { CmsConfig, I18nConfig, provideConfig } from \\"@spartacus/core\\";

@NgModule({
  declarations: [],
  imports: [
    AsmRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [ASM_FEATURE]: {
        module: () =>
          import('@spartacus/asm').then((m) => m.AsmModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: asmTranslations,
      chunks: asmTranslationChunksConfig,
    },
  })
  ]
})
export class AsmFeatureModule { }
"
`;
